indico/modules/events/management/templates/_settings.html

```diff
@@ -216,7 +216,35 @@
 {% endmacro %}
 
 
-{% macro render_event_settings(event, has_reference_types, has_event_labels, section=none, with_container=true) %}
+{% macro render_section_languages(event, languages, with_container=true) %}
+    {% call render_section(event, 'languages', 'icon-bubble-quote', '.edit_languages', _('Edit supported languages'),
+                           with_container=with_container) %}
+        {% set supported_languages = [] %}
+        {% for language in event.supported_languages %}
+            {% set name, territory, ambiguous = languages.get(language) %}
+            {% set language -%}
+                {{- name -}}
+            {%- if ambiguous and territory %} ({{ territory }}){% endif -%}
+            {%- endset %}
+            {{- supported_languages.append(language) or '' -}}
+        {% endfor %}
+        {% set name, territory, ambiguous = languages.get(event.default_language, ('', '', false)) %}
+        {% set default_language -%}
+            {{- name -}}
+        {%- if ambiguous and territory %} ({{ territory }}){% endif -%}
```

Comment:
Please move this into python code, it's way too much logic for a template. Usually modifying mutable objects in templates is a very good indicator for this.

Reply:
Hi @ThiefMaster, I have moved this code to the Event model, pl. suggest if it needs improvement?
In the RH, there would have been a duplicated code in two RHs, `RHEventSettings`, and `RHEditEventDataBase`